anta/tests/system.py

```diff
@@ -566,3 +566,98 @@ def test(self) -> None:
                 self._check_supervisor(output, supervisor_name="Standby")
             except (KeyError, JSONDecodeError, TypeError) as exc:
                 self.result.is_failure(f"Standby Supervisor - Failed to parse command output - {exc!s}")
+
+
+class VerifyFilePresence(AntaTest):
+    """Verifies the file presence on the flash drive.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the device is not under or entering maintenance.
+    * Failure: The test will fail if the device is under or entering maintenance.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.system:
+      - VerifyFilePresence:
+          filename: script.py
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["system"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaTemplate(template="dir {flash_memory}", revision=1, ofmt="text")]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyFilePresence test."""
+
+        filename: str
+        """Name of the file, including its extension (e.g., 'report.txt')."""
```

Comment:
We have an example already in the test docstring.

```suggestion
        """Name of the file, including its extension if any."""
```